Articles/Blog/PracticesPerformancePortableEco.md

```diff
@@ -0,0 +1,75 @@
+
+### Branching model
+
+Projects in the Kokkos Ecosystem reside on [GitHub](www.github.com/kokkos) and employ a branching process that looks similar to the [Git Flow workflow]( https://nvie.com/posts/a-successful-git-branching-model/):
+
+- The master branch is default, production-ready code, and held frozen between releases. 
+- The develop branch tracks contributions by developers, such as bug fixes, new features, enhancements, etc. Completed features are submitted as pull requests to the develop branch for code review and acceptance testing. GitHub’s issue tracker is used for communication regarding requests, bug reports, and progress.
+- Release cycles last approximately three to four months, at which point a release candidate branch is created off of develop followed by extensive testing, including integration testing with frameworks such as [Trilinos](https://trilinos.github.io/), prior to merge of the branch into the main branch and a tagged release.
+
+This process, coupled with testing requirements described in more detail next, aims to balance developer productivity with code stability – the compilers, hardware and configurations most widely used are shielded by acceptance testing of pull requests to the develop branch and leaving it in a relatively stable state for users eager to incorporate the newest features and capabilities. This process is also somewhat unusual for Git based libraries where the default master branch is updated more frequently. Release branches could also be used in similar way. However, we choose this approach so the default branch is a stable master that has been through integration testing with selected applications.
+
+### Testing, testing, testing
+
+As a performance-portable software library we are expected to support a variety of compilers and versions, hardware, and library configurations, for example with the Kokkos Ecosystem regular testing includes (at the time of writing):
+
+- Compilers: GCC (5.3.0 – 10); Clang (4.0-11); Intel (17-19); NVCC (9.2-11); ROCm/HIP (4.2); OneAPI (sycl-nightly tag); IBM/XL (16.1.1)
+- Hardware: x86_64 (various), KNL, ARMv8 ThunderX2, NVIDIA (K80, P100, V100, A100), AMD (Zen CPU, Vega MI25, MI50, MI60, MI100)
+- Backends: Serial, OpenMP, Pthreads, Cuda, HIP (Experimental), OpenMPTarget (Experimental), Sycl (Experimental), HPX
+
+Notice that we test several experimental options and support several older compilers that are years old in order to address the trade-off mentioned above. Total testing coverage requires hours of testing across several computing systems. It would be an unreasonable burden and detriment to coder productivity to require testing the full combinatorial set of supported builds and configurations. To balance developer productivity while preserving software quality and stability, testing of the code base consists of two stages:
```

Comment:
The end of this sentence says *two* stages, but discussion of three different types of testing follow.  Should it be *three* stages?  Or do you want to distinguish one (release testing?) in some way?

Reply:
Thanks for revision suggestion, addressed by 1a4586007f398b110bd1798dfb1e3ba12325aa18